src/python/web/security.py

```
@@ -0,0 +1,242 @@
+        if not request_t or origin_t != request_t:
+            raise bottle.HTTPError(403, "CSRF validation failed: origin mismatch")
+
+
+# ---------------------------------------------------------------------------
+# Rate Limiting
+# ---------------------------------------------------------------------------
+
+class _RateLimiter:
+    """Per-IP sliding-window rate limiter (in-memory)."""
+
+    def __init__(self, max_requests: int = 120, window_seconds: int = 60):
```

Ruff (0.15.2)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)